test/fetch/xhr_abort.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+#include <emscripten/fetch.h>
+
+void handleSuccess(emscripten_fetch_t *fetch)
+{
+  assert(false && "Should not succeed");
+  emscripten_fetch_close(fetch);
+}
+
+void handleError(emscripten_fetch_t *fetch)
+{
+  printf("handleError: %d %s\n", fetch->status, fetch->statusText);
+  bool isAbortedStatus = fetch->status == (unsigned short) -1;
+  assert(isAbortedStatus); // should have aborted status
+  if (isAbortedStatus)
```

Comment:
Seems odd to have an assert followed by this `if` check.. can't see assume its true after the abort?